### PR TITLE
Added toggle for virtual total level

### PR DESCRIPTION
**Pull request by robinwithes**
Closes https://github.com/runelite/runelite/issues/5878

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsConfig.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018, Robin Withes
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsConfig.java]: This should have an email or github link with it.

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsConfig.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup("virtuallevelsconfig")
+public interface VirtualLevelsConfig extends Config
+{
+	@ConfigItem(
+		keyName = "alterTotalLevel",
+		name = "Alter total level to reflect your virtual levels",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsConfig.java]: This is way too wordy. How about "Show virtual total level"?

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 3]
```diff
@@ -45,13 +47,23 @@
 public class VirtualLevelsPlugin extends Plugin
 {
 	private static final String TOTAL_LEVEL_TEXT_PREFIX = "Total level:<br>";
+	private static final String VIRT_TOTAL_LEVEL_TEXT_PREFIX = "Virtual total level:<br>";
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: Is there enough room in the total level box to show this text? Can you show a screenshot of this?

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 4]
```diff
@@ -102,10 +114,33 @@ public void onScriptCallbackEvent(ScriptCallbackEvent e)
 						continue;
 					}
 
-					level += Experience.getLevelForXp(client.getSkillExperience(s));
+					if (config.alterTotalLevel())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: What is this code doing? The proper behavior for when this config is disabled is to not change the script behavior, not to do some other code.

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 5]
```diff
@@ -102,10 +114,33 @@ public void onScriptCallbackEvent(ScriptCallbackEvent e)
...
+					{
+						level += Experience.getLevelForXp(client.getSkillExperience(s));
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: Store this `skillLevel` var locally since you will use it multiple times afterwards

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 6]
```diff
@@ -102,10 +114,33 @@ public void onScriptCallbackEvent(ScriptCallbackEvent e)
...
+					}
+					else
+					{
+						if (Experience.getLevelForXp(client.getSkillExperience(s)) > 99)
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: This whole if/else statement can be replaced with `level += Math.min(skillLevel, 99);`

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 7]
```diff
@@ -102,10 +114,33 @@ public void onScriptCallbackEvent(ScriptCallbackEvent e)
...
+						{
+							level += Experience.getLevelForXp(client.getSkillExperience(s));
+						}
+					}
+
+
 				}
 
-				stringStack[stringStackSize - 1] = TOTAL_LEVEL_TEXT_PREFIX + level;
+				if (config.alterTotalLevel())
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: Also this could be moved inside the other if statement above

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java (hunk 3 above)
Comment (robinwithes) [reply runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: https://gyazo.com/01e01bbeeb7662ef85cc723d7012e69e = current version,
https://gyazo.com/f349a93a279f34cffb4e5d778da91549 = another one i tried.
Both of them clip out a little bit, im not sure what would fit better though.
Maybe just leave show total level anyway? Not sure..

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: Yeah, it seems like it would be better to leave the text be.

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 8]
```diff
@@ -93,19 +115,22 @@ public void onScriptCallbackEvent(ScriptCallbackEvent e)
...
+						{
+							continue;
+						}
 
-					level += Experience.getLevelForXp(client.getSkillExperience(s));
-				}
+						level += Experience.getLevelForXp(client.getSkillExperience(s));
 
-				stringStack[stringStackSize - 1] = TOTAL_LEVEL_TEXT_PREFIX + level;
+					}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: This should be moved to the line above it to maintain the empty line between the loop and setting of the stringStack.

**Comment by Nightfirecat** [comment]
Looks much better now.

**Comment by robinwithes** [comment]
> Looks much better now.

Thanks for all the feedback!

File: runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java [hunk 9]
```diff
@@ -44,14 +49,23 @@
 )
 public class VirtualLevelsPlugin extends Plugin
 {
-	private static final String TOTAL_LEVEL_TEXT_PREFIX = "Total level:<br>";
+	private static final String VIRT_TOTAL_LEVEL_TEXT_PREFIX = "Total level:<br>";
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java]: Is there any particular reason for this variable rename?

**Comment by Nightfirecat** [comment]
Looks like a merge went wrong here?

**Comment by xDemoN** [comment]
Hmm, it looks like this was superseded by #8676

**Comment by robinwithes** [comment]
well yes, this is from 2018
